File: MVVM/app/src/main/java/jp/ogiwara/java/mvvm/MainActivity.java
package jp.ogiwara.java.mvvm;

import android.databinding.DataBindingUtil;
import android.databinding.ObservableInt;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import org.jetbrains.annotations.NotNull;

import jp.ogiwara.java.mvvm.contract.RepositoryListViewContract;
import jp.ogiwara.java.mvvm.databinding.ActivityMainBinding;
import jp.ogiwara.java.mvvm.model.GitHubService;
import jp.ogiwara.java.mvvm.view.MVVMApplication;
import jp.ogiwara.java.mvvm.view_model.RepositoryListViewModel;

public class MainActivity extends AppCompatActivity implements RepositoryListViewContract{

    private ActivityMainBinding binding = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this,R.layout.activity_main);

        GitHubService gitHubService = ((MVVMApplication) getApplication()).getGitHubService();
        binding.setViewModel(new RepositoryListViewModel((RepositoryListViewContract) this, gitHubService));
    }

    @Override
    public void showError(@NotNull Throwable e){
        e.printStackTrace();
        Toast.makeText(this,"ERROR! CHECK LOG.",Toast.LENGTH_LONG);
    }

    @Override
    public void showRepositories(@NotNull GitHubService.Repositories repositories) {
        binding.repositories.setText(repositories.toString());
    }
}

File: MVP/app/src/main/java/jp/ogiwara/java/mvp/MainActivity.kt
package jp.ogiwara.java.mvp

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import butterknife.bindView
import jp.ogiwara.java.mvp.contract.UserActions
import jp.ogiwara.java.mvp.contract.View
import jp.ogiwara.java.mvp.presenter.RepositoryLoader
import jp.ogiwara.java.mvp.view.MVPApplication

class MainActivity : AppCompatActivity(),View {

    val repositories: TextView by bindView(R.id.repositories)
    val loadButton: Button by bindView(R.id.load_button)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //region load views

        val gitHubService = (application as MVPApplication).githubService
        val repositoryLoader = RepositoryLoader(this as View,gitHubService!!)

        loadButton.setOnClickListener {//これがMVVMだとなくなる
            repositoryLoader.loadRepository()
        }
        //endregion
    }

    override fun showRepositories(str: String) {
        repositories.text = str
    }
}
